### PR TITLE
final project idea

### PR DESCRIPTION
**Pull request by tamanduaa**

### CODE REVIEW

**Comment by jamdon** [comment]
Hi Lauren,

Only you can know how difficult it might be as I'm not familiar with the framework you're going to use. My main advice is to think now in terms of how you'll incorporate the required elements of the project.

**Comment by jamdon** [comment]
Hi Lauren,

Just wanted to clarify if this is your final submission and you want it graded, if you want feedback on it so you can work on it more, or if you have more work incoming later today.

**Comment by tamanduaa** [comment]
Hi James,

I would love feedback on this if you can.  I'm still working on it but I am
a bit stuck so any feedback or direction on how to properly set up backbone
would be great.  I am also trying to connect it so that the links at the
bottom of the page link up to my props list not the food trucks list.  I
have just made another commit.

Thank you so much!
Lauren

**Comment by jamdon** [comment]
The first thing you need is to add the proper dev dependencies to your package.json file. If you're going to use grunt, then you have to install (and save to your package.json file in the dev dependencies) the grunt plugins that the Gruntfile is specifying so it can build your client script bundles.

**Comment by tamanduaa** [comment]
Thank you so much James! I have installed the grunt plugins. I am trying to
add a link at the bottom that will work the way that the food trucks
example sorts trucks by date or type.  But in this example it would sort my
list of props by material and due date.  I am a bit confused about how to
make those links reference my list.  I have been trying to work on the
Props List one.  It goes to the right URL but displays nothing.  How can I
connect it to the right data?

Thank you,
Lauren

File: index.js [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+app.use(express.static('public'));
+//app.use(bodyParser.urlencoded({ extended: false }));
+
+app.use('/trucks', truckRouter);
+app.use('/food-types', foodTypeRouter);
+app.use('/food-trucks', foodTypeRouter);
+app.use('/today', foodTypeRouter);
+app.use('/props', foodTypeRouter);
+
+app.use(express.static(root));
```
Comment (jamdon) [inline index.js:27]: If you're going to use root as a variable here, then you have to define it first.

**Comment by jamdon** [comment]
Ok, I was able to compile your client-side files, but I can't run your server yet. I left a comment inline.

**Comment by tamanduaa** [comment]
Thank you James!  I have defined root and commented out the line:

app.use(express.static('public'));

because I'm guessing that conflicts with

app.use(express.static(root));

Are you able to run the server now?

Thank you!

Lauren

**Comment by jamdon** [comment]
You're not getting any data back when you try to fetch in client.js because you're making a call to the /trucks endpiont, but your express router is mounted on the /api/trucks route. Also, just so I'm clear, it doesn't appear you actually use any Backbone components in your client side application. Is that correct?

**Comment by tamanduaa** [comment]
Thank you James!  I've updated it to make a call to /trucks instead and
changed the links so it appears they are going to the correct url now.  I'm
in the process of updating the view files.  Once I add view files that
display my props list and filter it, I think that would mean I am using
Backbone components in my client side app.  Am I correct in thinking that?

Thank you,
Lauren

**Comment by jamdon** [comment]
I see the files, but nothing is ever used. Your index.html never loads your main.bundled.js or your templates.js. Was it your intention to use any of it?

**Comment by tamanduaa** [comment]
Hi James,

Ahh okay I see that.  I've added them to index.html.  My intention was to
use templates to write out a filtered list of props, but I ran out of time
to write the code to filter them.  I wanted to have a button to show all
props that used a certain material.  So I would display a list of the props
whose entry for that material was greater than zero.  Ideally that list
would show the prop name and the material amounts, and I thought
templates.js would be needed to do that.

Thank you,
Lauren

**Comment by tamanduaa** [comment]
Also I realize time is up because the project is due today.  In order to
submit it do I need to do a pull request?

Thank you,
Lauren

**Comment by jamdon** [comment]
No, this is your pull request. You've just been updating it. 👍

**Comment by tamanduaa** [comment]
Got it, thank you so much for all your help James!

**Pull request closed without merging**